ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/LedgerTimeValidator.scala

```diff
@@ -46,68 +46,26 @@ private[transaction] class LedgerTimeValidator(defaultConfig: Configuration)
                 _ => StepContinue(transactionEntry),
               )
           case None => // Pre-execution: propagate the time bounds and defer the checks to post-execution.
-            val maybeDeduplicateUntil =
-              getLedgerDeduplicateUntil(transactionEntry, commitContext)
-            val minimumRecordTime = transactionMinRecordTime(
-              transactionEntry.submissionTime,
-              transactionEntry.ledgerEffectiveTime,
-              maybeDeduplicateUntil,
-              timeModel,
-            )
-            val maximumRecordTime = transactionMaxRecordTime(
-              transactionEntry.submissionTime,
-              transactionEntry.ledgerEffectiveTime,
-              timeModel,
-            )
-            commitContext.deduplicateUntil = maybeDeduplicateUntil
-            commitContext.minimumRecordTime = Some(minimumRecordTime)
-            commitContext.maximumRecordTime = Some(maximumRecordTime)
-            val outOfTimeBoundsLogEntry = DamlLogEntry.newBuilder
-              .setTransactionRejectionEntry(
-                rejections.preExecutionOutOfTimeBoundsRejectionEntry(
-                  transactionEntry,
-                  minimumRecordTime,
-                  maximumRecordTime,
+            (commitContext.minimumRecordTime, commitContext.maximumRecordTime)
+              .mapN((minimumRecordTime, maximumRecordTime) => {
```

Comment:
Why are we introducing cats here? We aren't using this library anywhere in kvutils or under the broader `com.daml.ledger.participant.state` package.